APMrover2/APMrover2.cpp

```diff
@@ -195,9 +195,13 @@ void Rover::ahrs_update()
  */
 void Rover::gcs_failsafe_check(void)
 {
-    if (g.fs_gcs_enabled) {
-        failsafe_trigger(FAILSAFE_EVENT_GCS, last_heartbeat_ms != 0 && (millis() - last_heartbeat_ms) > 2000);
+    if (!g.fs_gcs_enabled) {
+        // gcs failsafe disabled
+        return;
     }
+
+    // check for updates from GCS within 2 seconds
+    failsafe_trigger(FAILSAFE_EVENT_GCS, failsafe.last_heartbeat_ms != 0 && (millis() - failsafe.last_heartbeat_ms) > 2000);
```

Comment:
This still has the hardcoded 2 second timeout.

Reply:
In the case you're testing though it's the time of the throttle failsafe that is important.  That's been reduced from 2 seconds to 1 second (+FS_TIMEOUT value).  We can potentially make it shorter still though.